### PR TITLE
Install script

### PR DESCRIPTION
**Pull request by edisonhello**
Create a script provides automatic installation. The script was tested in Ubuntu 16.04, 18.04 and 20.04 by confirming `curl localhost` returns the TIOJ home page.

### CODE REVIEW

**Comment by oToToT** [comment]
Maybe you could also change tioj-docker after this PR merged?

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  database: tioj_production
+EOF
+
+TOKEN=$(rake secret)
+KEY=$(rake secret)
+
+sed "s/\".*\"/\"$TOKEN\"/" $TIOJ_PATH/config/initializers/secret_token.rb.example > $TIOJ_PATH/config/initializers/secret_token.rb
+sed "s/\".*\"/\"$KEY\"/" $TIOJ_PATH/config/initializers/fetch_key.rb.example > $TIOJ_PATH/config/initializers/fetch_key.rb
+
+sed "s,l.*,l = $URL,; s/y.*/y = $KEY/" $MIKU_PATH/app/tioj_url.py.example > $MIKU_PATH/app/tioj_url.py
```
Comment (edisonhello) [inline setup.sh:110]: Surround URL and KEY with `"`.

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+TIOJ_PATH=$HOME/tioj
+MIKU_PATH=$HOME/miku
+DB_USERNAME=root
+DB_PASSWORD=SAMPLE_PASSWORD
```
Comment (edisonhello) [inline setup.sh:6]: Maybe prompt for password.

File: setup.sh [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+sed "s,l.*,l = $URL,; s/y.*/y = $KEY/" $MIKU_PATH/app/tioj_url.py.example > $MIKU_PATH/app/tioj_url.py
+
+cd $TIOJ_PATH
+RAILS_ENV=production rake db:create db:schema:load db:seed
+RAILS_ENV=production rake assets:precompile
+mkdir public/announcement
+echo -n '{"name":"","message":""}' > public/announcement/anno
+
+sudo make -j -C $MIKU_PATH
+
```
Comment (edisonhello) [inline setup.sh:119]: Add more about starting web server and judge server.
image

**Review by edisonhello — COMMENTED** [review]
Request changes.

**Comment by adrien1018** [comment]
Already rewritten.

**Pull request closed without merging**